kore/src/Kore/Logger/CriticalExecutionError.hs

```diff
@@ -0,0 +1,39 @@
+{- |
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+
+-}
+
+module Kore.Logger.CriticalExecutionError
+    ( CriticalExecutionError
+    , criticalExecutionError
+    ) where
+
+import Control.Monad.Catch
+    ( SomeException
+    , displayException
+    )
+import Data.Text.Prettyprint.Doc
+    ( Pretty (..)
+    )
+
+import Kore.Logger
+
+newtype CriticalExecutionError =
+    CriticalExecutionError
+    { executionError :: SomeException
+    }
+
+instance Pretty CriticalExecutionError where
+    pretty CriticalExecutionError { executionError } =
+        pretty . displayException $ executionError
+
+instance Entry CriticalExecutionError where
```

Comment:
Could you explain me why did you add this instance?

Reply:
@andreiburdusa When we want to log something new, we add a new type which implements the `Entry` class so that we can log it (see https://github.com/kframework/kore/blob/504d615fddb8a3705de36337a78e542a61710a9d/kore/src/Kore/Logger.hs#L243). The reason we make new types for different logging information (instead of just using `LogMessage`, see the functions here https://github.com/kframework/kore/blob/504d615fddb8a3705de36337a78e542a61710a9d/kore/src/Kore/Logger.hs#L148) is that we can use the types themselves to provide additional information in the log and to filter what gets printed. That's my understanding of our logger.